org.eclipse.zest.core/src/org/eclipse/zest/core/widgets/internal/ZestRootLayer.java

```diff
@@ -64,37 +64,61 @@ public class ZestRootLayer extends FreeformLayer {
 	 */
 	private boolean isLayerKnown = false;
 
+	protected int getConnectionsLayer() {
+		return CONNECTIONS_LAYER;
+	}
+
+	protected int getSubgraphsLayer() {
+		return SUBGRAPHS_LAYER;
+	}
+
+	protected int getNodesLayer() {
+		return NODES_LAYER;
+	}
+
+	protected int getConnectionsHighlightedLayer() {
+		return CONNECTIONS_HIGHLIGHTED_LAYER;
+	}
+
+	protected int getNodesHighlightedLayer() {
+		return NODES_HIGHLIGHTED_LAYER;
+	}
+
+	protected int getTopLayer() {
+		return TOP_LAYER;
+	}
+
 	/**
 	 * Adds a node to the ZestRootLayer
 	 *
 	 * @param nodeFigure The figure representing the node
 	 */
 	public void addNode(IFigure nodeFigure) {
-		addFigure(nodeFigure, NODES_LAYER);
+		addFigure(nodeFigure, getNodesLayer());
```

Comment:
I'll be honest; I don't like this at all. A `getNodesLayer()` method should always return `NODES_LAYER`. Anything else I would consider a conceptual flaw.

Can't you just overwrite the `addNode(IFigure)` method and then use a different constant? For example:
```java
public void addNode(IFigure nodeFigure) {
	addFigure(nodeFigure, CONNECTIONS_LAYER)
}
```

You might have to make the `changeFigureLayer()` method protected to overwrite the highlight methods, but that's fine. It still looks a little weird adding figures to other layers, but at least it's immediately obvious what's happening, when looking at it.

Reply:
> Can't you just overwrite the addNode(IFigure) method and then use a different constant?

Oh... yes that makes a lot more sense. I might have been a bit blinded by the constants, that I missed I don't even have to use them.
Thank you. I will close this PR.